src/v1_0/rw/elog.cpp

```diff
@@ -0,0 +1,140 @@
+    {
+        std::vector<Poco::XML::Node *> domains_xml = xmlFindNodes(rws_result, XMLAttribute("class", "elog-domain-li"));
+        std::vector<std::pair<int, std::string>> elogDomains;
+        for (auto node : domains_xml)
+        {
+            std::pair<int, std::string> domain;
+            domain.first = std::stoi(xmlNodeGetAttributeValue(node, "title"));
+            domain.second = xmlFindTextContent(node, XMLAttribute("class", "domain-name"));
+            elogDomains.push_back(domain);
+        }
+        return elogDomains;
+    }
+
+    std::vector<std::pair<int, std::string>> getElogDomains(RWSClient &client, std::string lang)
+    {
+        std::stringstream uri;
+        uri << "/rw/elog?lang=" << lang << "&resource=count";
+        POCOResult poco_result = client.httpGet(uri.str());
+        RWSResult const &rws_result = parseXml(poco_result.content());
+        return parseDomains(rws_result);
+    }
+
+    ElogMessage parseElogMessageXml(std::string const xml_string)
+    {
+        Poco::AutoPtr<Poco::XML::Document> rws_result = parseXml(xml_string);
+        ElogMessage message;
+        message.messageType = static_cast<ElogMessageType>(std::stoi(xmlFindTextContent(rws_result, XMLAttribute("class", "msgtype"))));
+        message.code = std::stoi(xmlFindTextContent(rws_result, XMLAttribute("class", "code")));
+        message.sourceName;
+        std::istringstream ss(xmlFindTextContent(rws_result, XMLAttribute("class", "tstamp")));
+        ss >> std::get_time(&message.timestamp, "%Y-%m-%d T %H:%M:%S");
+        message.title = xmlFindTextContent(rws_result, XMLAttribute("class", "title"));
+        message.desc = xmlFindTextContent(rws_result, XMLAttribute("class", "desc"));
+        message.conseqs = xmlFindTextContent(rws_result, XMLAttribute("class", "conseqs"));
+        message.causes = xmlFindTextContent(rws_result, XMLAttribute("class", "causes"));
+        message.actions = xmlFindTextContent(rws_result, XMLAttribute("class", "actions"));
+
+        int argc = std::stoi(xmlFindTextContent(rws_result, XMLAttribute("class", "argc")));
+
+        for (int i = 1; i <= argc; i++)
+        {
+            std::stringstream arg_name;
+            arg_name << "arg" << i;
+            auto node = xmlFindNodes(rws_result, XMLAttribute("class", arg_name.str()))[0];
+
+            std::string value = node->innerText();
+            std::string valueType = xmlNodeGetAttributeValue(node, "type");
+
+            message.argv.emplace_back(
+                std::make_pair(
+                    arg_name.str(),
+                    makeElogMessageArg(valueType, value)));
+        }
+        return message;
+    }
+
+    ElogMessage getElogMessage(RWSClient &client, int const domain, int const seqnum, std::string const &lang)
+    {
+        std::stringstream uri;
+        uri << "/rw/elog/" << static_cast<int>(domain) << "/" << seqnum << "?lang=" << lang;
```

Comment:
`domain` is already an `int`
```suggestion
        uri << "/rw/elog/" << domain << "/" << seqnum << "?lang=" << lang;
```